File: CFCnfrmEntrg/app/src/main/java/com/example/cfcnfrmentrg/MainActivity.java
package com.example.cfcnfrmentrg;

import android.app.ProgressDialog;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import com.google.gson.Gson;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;

public class MainActivity extends AppCompatActivity {
    private RequestQueue queue;
    EditText firstNumEditText;
    TextView resultTextView;
    Button addBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        queue= Volley.newRequestQueue(this);

        firstNumEditText =(EditText) findViewById(R.id.firstNumEditText);
        resultTextView=(TextView) findViewById(R.id.resultTextView);

        addBtn = (Button) findViewById(R.id.addBtn);

        addBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //int num1=Integer.parseInt(firstNumEditText.getText().toString());
                //int num2=Integer.parseInt(secondNumEditText.getText().toString());
                obtenerDatosVolley();
            }
        });
    }//fin void on create

    //________________________




    //------------------------------------------------------------------------------------
    private void obtenerDatosVolley()
    {
        //20190610 obtendiendo datos desde los https funciona
        //String url="https://api.androidhive.info/contacts/";
        //String url="http://jsonplaceholder.typicode.com/todos/1";
        //ninguno de estos locales funciona
        //String url="http://10.1.0.136/CFService/Service1.svc/GetData/xr";//varios resulados
        //String url="http://10.1.0.136/CFService/Service1.svc/GetData2/"+firstNumEditText.getText().toString();//un resultados
        //test update
        String url="http://10.1.0.136/CFService/Service1.svc/ConfirmUser/"+firstNumEditText.getText().toString();//un resultados



        JsonObjectRequest request = new JsonObjectRequest(Request.Method.GET, url, null, new Response.Listener<JSONObject>() {
            @Override
            public void onResponse(JSONObject response)
            {
                try
                {
                    //varios resultados
                    //JSONArray mJSONArray=response.getJSONArray("contacts");
                    //JSONObject mJSONObject = mJSONArray.getJSONObject(0);
                    //String name = mJSONObject.getString("name");

                    //un reusltado
                    //String name=response.getString("title");

                    //test update
                    String name=response.getString("Result");

                    resultTextView.setText(name);
                    Toast.makeText(MainActivity.this,"result:"+response,Toast.LENGTH_LONG).show();


                } catch (JSONException e) {
                    resultTextView.setText("error2");
                    e.printStackTrace();

                }

            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error)
            {
                resultTextView.setText("error3");

            }
        });

        queue.add(request);


    }//Fin obtener Datos Volley

    //-------------------------------------------------------------------------------------


}

File: Procesos/10.1PostJSClass/app/src/main/java/com/example/a101postjsclass/MainActivity.kt
package com.example.a101postjsclass

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONException
import org.json.JSONObject


//Post solo para verificar el envio de kotlin al servicio

//si funcioa proframar completo el kt para crear entrega
//poniendo los datos en el texview pa no matarse con esos adapter__mientras le agarras a los adapters
class MainActivity : AppCompatActivity() {
    private var queue: RequestQueue? = null

    //Json4Kotlin_Base xJson4Kotlin_Base=new Json4Kotlin_Base();
//val users: List<User> = listOf( User("Tom", 32), User("John", 64) ) !!!

    //asignando valores manual mente para mandar el post de un solo
    //creamdo propiedades de Base
    var testlist= Posttest("10","001810")
    var listposttest:List<Posttest> = mutableListOf(testlist,Posttest("5","002010"))

    //creando base
    var Base=Json4Kotlin_Base(360010,"PosteandoTest",listposttest,0)


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnAgregarCodigo.setOnClickListener{

            //anadiendo articulos
            Base.posttest=Base.posttest+Posttest(txtcantidad.text.toString(),txtcodigo.text.toString())
        }

        //enviar datos volley
        btnCerrar.setOnClickListener{
            obtenerDatosVolley()
        }




    }

    //------------------------------------------------------------------------------------
    private fun obtenerDatosVolley() {
        //20190610 obtendiendo datos desde los https funciona
        //String url="https://api.androidhive.info/contacts/";
        //String url="http://jsonplaceholder.typicode.com/todos/1";
        val url="http://localhost:3397/HouseService.svc/ListPostTest"//varios resulados
        //val url = "http://10.1.0.136/CFService/Service1.svc/GetData2/xr"//un resultados
        var gson = Gson()
        var jsonString = gson.toJson(Base)
        val jsonObject = JSONObject(jsonString)

        val request = JsonObjectRequest(Request.Method.POST, url, jsonObject,
            Response.Listener { response ->
                try {


                    val mJSONArray = response.getJSONArray("Posttest")
                    val mJSONObject = mJSONArray.getJSONObject(0)


                    val name1 = response.getString("message")
                    var name2 = mJSONObject.getString("status")
                    //OPCION DIRECTA
                    //val name2 = response.getJSONArray("users").getJSONObject(0).getString("nombre")


                    //un reusltado
                    //val name = response.getString("title")


                    val topic = Gson().fromJson(response.toString(), Json4Kotlin_Base::class.java  )

                    name2 = topic.posttest[1].item

                    textView.setText("mesaage: $name1 nombre: $name2")
                    Toast.makeText(this@MainActivity, "Nombre:$name1 Nombre2:$name2", Toast.LENGTH_LONG).show()


                } catch (e: JSONException) {
                    textView.setText("error2")
                    e.printStackTrace()
                }
            }, Response.ErrorListener { textView.setText("error3") })

        queue?.add(request)

    }//Fin obtener Datos Volley

    //-------------------------------------------------------------------------------------

}//Fin Mainactiviy
